sql: flaky UDF tracing test

pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test.TestLogic_udf [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/10812147?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/10812147?buildTab=artifacts#/) on master @ [c92599f8fc4f2191c6f9b3d5f0d2bc8d0009305b](https://github.com/cockroachdb/cockroach/commits/c92599f8fc4f2191c6f9b3d5f0d2bc8d0009305b):


```
=== RUN   TestLogic_udf
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/14e536c19bd6f7f614acc0eb4eba225a/logTestLogic_udf1668289614
    test_log_scope.go:81: use -show-logs to present logs inline
[01:21:37] setting distsql_workmem='99147B';
    logictestbase.go:488: statement/query skipped with reason: unsupported configuration local-mixed-22.2-23.1, statement/query only supports local-legacy-schema-changer (no issue given)
=== CONT  TestLogic_udf
    logic.go:4225: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/14e536c19bd6f7f614acc0eb4eba225a/logTestLogic_udf1668289614
--- FAIL: TestLogic_udf (91.22s)
=== RUN   TestLogic_udf/execution
    logic.go:2906: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7257/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/udf:2181: SELECT message FROM [SHOW TRACE FOR SESSION] WHERE message ~ 'udf'
        expected:
            === SPAN START: udf-stmt-trace_fn-1 ===
            === SPAN START: udf-stmt-trace_fn-2 ===
            === SPAN START: udf-stmt-trace_fn-1 ===
            === SPAN START: udf-stmt-trace_fn-2 ===
            === SPAN START: udf-stmt-trace_fn-1 ===
            === SPAN START: udf-stmt-trace_fn-2 ===
            
        but found (query options: "rowsort" -> ignore the following ordering of rows) :
            === SPAN START: udf-stmt-trace_fn-1 ===
            === SPAN START: udf-stmt-trace_fn-2 ===
            === SPAN START: udf-stmt-trace_fn-1 ===
            === SPAN START: udf-stmt-trace_fn-2 ===
            === SPAN START: udf-stmt-trace_fn-1 ===
            plan diagram URL:
            https://cockroachdb.github.io/distsqlplan/decode.html#eJyMkUFP4zAQhe_7K6I57UrObtKuaeUrLVIlSlHbG4qqqT0pFm4cPA4FVfnvKAkIxKnHee95vnnyGfjZgYLN_HZ-vU3qZu-s_hsDatqV1W_8k9ysV8tkECLuQUDlDd3hkRjUA-RQCKiD18TsQyed-8DCvILKBNiqbmInFwK0DwTqDNFGR6DAeY0ueUHXECfZvwwEGIpoXZ9vBfgmfr3miAcClX_DLWagslZcTlwT175iuoiU_SCleVsIIHOgoSb7Jmi6D1732WFc9Yt6wRDHwc2HYVF9WhwD4XE4vxBQOn_aWQMK9hM5womZpmUudfp_LGU6vZKjdKLHcmxIS03dF5QOD9wV2zz6U792-1Z3Z5XomAQs8YlmFCkcbWU5Wv3htO2v9wAAAP__aKui6g==
            === SPAN START: udf-stmt-trace_fn-2 ===
            
[01:23:06] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7257/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/udf: 351 statements
    logic.go:2164: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7257/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/udf:2191: too many errors encountered, skipping the rest of the input
[01:23:06] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7257/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-22.2-23.1/local-mixed-22_2-23_1_test_/local-mixed-22_2-23_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/udf with config local-mixed-22.2-23.1: 351 tests, 2 failures
[01:23:07] --- total progress: 351 statements
--- total: 351 tests, 2 failures
    --- FAIL: TestLogic_udf/execution (2.35s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_udf.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-29528

--- a/pkg/sql/logictest/testdata/logic_test/udf
+++ b/pkg/sql/logictest/testdata/logic_test/udf
@@ -2178,7 +2178,7 @@ statement ok
 SET tracing = off
 
 query T rowsort
-SELECT message FROM [SHOW TRACE FOR SESSION] WHERE message ~ 'udf'
+SELECT message FROM [SHOW TRACE FOR SESSION] WHERE message ~ '^=== SPAN START: udf-stmt-trace_fn'
 ----
 === SPAN START: udf-stmt-trace_fn-1 ===
 === SPAN START: udf-stmt-trace_fn-2 ===

